File: 1209 - Remove All Adjacent Duplicates in String II/solution.py
class Solution:
    def removeDuplicates(self, s: str, k: int) -> str:
        stk = []
        for c in s:
            if stk and stk[-1][0] == c:
                stk[-1][1] += 1
                if stk[-1][1] == k:
                    stk.pop()
            else:
                stk.append([c, 1])
        return ''.join(c*i for c, i in stk)

File: 1209 - Remove All Adjacent Duplicates in String II/solution.cpp
class Solution {
public:
    string removeDuplicates(string s, int k) {
        int idx = 0, n = s.size();
        vector<int> cnt(n);
        for(int i = 0; i < n; i++, idx++){
            s[idx] = s[i];
            cnt[idx] = (idx>0&&s[idx-1]==s[idx])?cnt[idx-1]+1:1;
            if(cnt[idx] == k) idx -= k;
        }
        return s.substr(0, idx);
    }
};